src/Conversion/ONNXToTOSA/ONNXToTOSACommon.hpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//====------ ONNXToTOSACommon.hpp - ONNX dialects to TOSA lowering --------===//
+//
+// Copyright (c) 2022 Advanced Micro Devices, Inc.
+//
+// =============================================================================
+//
+// This file contains common code shared by the functions performing the
+// lowering to the TOSA dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/Quant/QuantTypes.h"
+#include "mlir/Dialect/Tosa/IR/TosaOps.h"
+
+#include "mlir/IR/MLIRContext.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+#include "src/Dialect/ONNX/DialectBuilder.hpp"
+#include "src/Dialect/ONNX/ONNXOps.hpp"
+#include "src/Dialect/ONNX/ONNXOpsHelper.hpp"
+#include "src/Pass/Passes.hpp"
+#include "src/Transform/ONNX/ConstPropHelper.hpp"
+
+//===----------------------------------------------------------------------===//
+// Functions to add lowering patterns for frontend operations.
+//===----------------------------------------------------------------------===//
+
+namespace onnx_mlir {
+
+//===----------------------------------------------------------------------===//
+// Check for valid TOSA types.
+//===----------------------------------------------------------------------===//
+
+inline bool isTOSASignedInt(Type type) {
```

Comment:
With `git diff origin/main origin/ferdinand.bring_upstream_onnxtotosa  -- src/Conversion/ONNXToTOSA/ test/mlir/tosa` I see diffs. Are these necessary? (I'd rather not have them as we have to maintain them when bumping and opening PRs upstream.)

```
--- a/src/Conversion/ONNXToTOSA/ONNXToTOSACommon.hpp
+++ b/src/Conversion/ONNXToTOSA/ONNXToTOSACommon.hpp
@@ -36,15 +36,15 @@ namespace onnx_mlir {
 // Check for valid TOSA types.
 //===----------------------------------------------------------------------===//
 
-inline bool isTOSASignedInt(mlir::Type type) {
-  mlir::IntegerType intType = type.dyn_cast<mlir::IntegerType>();
+inline bool isTOSASignedInt(Type type) {
+  IntegerType intType = type.dyn_cast<IntegerType>();
   std::set<unsigned> intWidth{8, 16, 32, 48, 64};
   return intType && intType.isSigned() &&
          (intWidth.find(intType.getWidth()) != intWidth.end());
 }
 
-inline bool isTOSAFloat(mlir::Type type) {
-  return type.isa<mlir::BFloat16Type, mlir::Float16Type, mlir::Float32Type>();
+inline bool isTOSAFloat(Type type) {
+  return type.isa<BFloat16Type, Float16Type, Float32Type>();
 }
 
 //===----------------------------------------------------------------------===//
@@ -59,6 +59,6 @@ template <typename Op>
 using TOSAOp = typename TOSADialectOp<Op>::Op;
 
 // `Math` directory methods:
-void populateLoweringONNXElementwiseOpToTOSAPattern(mlir::ConversionTarget &,
-    mlir::RewritePatternSet &, mlir::TypeConverter &, mlir::MLIRContext *);
+void populateLoweringONNXElementwiseOpToTOSAPattern(
+    ConversionTarget &, RewritePatternSet &, TypeConverter &, MLIRContext *);
 } // namespace onnx_mlir
```

Reply:
Answering for myself - there was a later commit that cleaned up namespaces. We don't need to take this just yet, so I'm happy with the PR as-is.